crates/shim/src/cgroup.rs

```diff
@@ -199,6 +193,34 @@ fn get_cgroup(pid: u32) -> Result<Cgroup> {
     Ok(cgroup)
 }
 
+/// Get the cgroups v2 path given a PID
+pub fn get_cgroups_v2_path_by_pid(pid: u32) -> Result<PathBuf> {
+    // todo: should upstream to cgroups-rs
+    let path = format!("/proc/{}/cgroup", pid);
+    let content = fs::read_to_string(path).map_err(io_error!(e, "read cgroup"))?;
+    let content = content.lines().next().unwrap_or("");
+
+    let Ok(path) = parse_cgroups_v2_path(content)?.canonicalize() else {
```

Comment:
is there any concern about resolving symlinks or relative paths via canonicalize here?

Reply:
In the go implementation the path is cleaned here https://github.com/opencontainers/runc/blob/1950892f69597aa844cbf000fbdf77610dda3a44/libcontainer/cgroups/fs2/defaultpath.go#L48

Reply:
Testing is harder if the actual path has to exist. Maybe we can just normalize it instead of canonicalizing it, although rust's stdlib doesnt have a normalize method, that would require a 3rd party crate.

Reply:
Or do something like this:
https://github.com/containers/youki/blob/0b89e1d52089c8c328496dab02dcc2bcf5e1b53c/crates/libcontainer/src/utils.rs#L92

Reply:
If I understand the Go's CleanPath correctly, it's like normalizing the path but without verifying it exists in the filesystem, unlick `fs::canonicalize()` right?

Reply:
I think the key is:
```
//  "This ensures that a
// path resulting from prepending another path will always resolve to lexically
// be a subdirectory of the prefixed path."
```

Reply:
> In the go implementation the path is cleaned here https://github.com/opencontainers/runc/blob/1950892f69597aa844cbf000fbdf77610dda3a44/libcontainer/cgroups/fs2/defaultpath.go#L48

@Burning1020 @mxpv does this path need be cleaned? It is there in the go code but I am not sure if I completely understand why it is needed. If not we should be ok to merge this.